mindmeld/auto_annotator.py

```diff
@@ -759,4 +770,63 @@ def _resolve_person(self, entity):
         return entity
 
 
+class BootstrapAnnotator(Annotator):
+    """ Bootstrap Annotator class used to generate annotations based on existing annotations.
+    """
+
+    def __init__(self, app_path, config=None):
+        super().__init__(app_path=app_path, config=config)
+        self.confidence_threshold = self.config.get("confidence_threshold", 0)
+        print("BootstrapAnnotator is loading", self.app_path)
+        self.nlp = NaturalLanguageProcessor(self.app_path)
+        self.nlp.build()
+
+    def parse(self, sentence, entity_types, domain: str, intent: str, **kwargs):
+        """
+        Args:
+                sentence (str): Sentence to detect entities.
+                entity_types (list): List of entity types to parse. If None, all
+                        possible entity types will be parsed.
+        Returns: entities (list): List of entity dictionaries.
+        """
+        er = self.nlp.domains[domain].intents[intent].entity_recognizer
+        predict_proba_output = er.predict_proba(sentence)
+        entities = []
+        for data in predict_proba_output:
+            entity, confidence = data
+            if (
+                not entity_types or entity.entity.type in entity_types
+            ) and confidence >= self.confidence_threshold:
+                entity = {
+                    "body": entity.text,
+                    "start": entity.span.start,
+                    "end": entity.span.end + 1,
+                    "dim": entity.entity.type,
+                    "value": entity.entity.value,
+                }
```

Comment:
Can the auto annotator support roles?

Reply:
Not currently but I am looking into how I could add support for it

Reply:
Added support for role labeling. Since the entity recognizer does not provide role classification information, nlp.process() for the specific domain and intent pair is used.